Holovibes/cuda_includes/reduce.cuh

```diff
@@ -5,42 +5,21 @@
 #pragma once
 using uint = unsigned int;
 
-/*! \brief Reduce operation gpu side
- *
- * \param input Input buffer
- * \param result Result of the reduce (even with double, imprecision may arise)
- * \param size Input size
- * \param reduce_op Operator used for the reduction
- * \param atomic_op Atomic operator used for the write back in result
- * \param identity_elt Identity element needed to initilize data (add needs 0, min needs max...)
- *
- * This kernel has been highly tuned in order to maximize the memory bandwidth usage
- * Numerous benches have been done to achieve the best result possible
- * Don't modify this kernel unless making benches
- */
-template <typename T, typename U, typename R_OP, typename A_OP>
-void reduce_generic(const T* const __restrict__ input,
-                    const uint size,
-                    const R_OP reduce_op,
-                    const A_OP atomic_op,
-                    const T identity_elt,
-                    const cudaStream_t stream);
-
```

Comment:
Reduce generic is not use anywhere in the project except in reduce.cuhxx so there is no point to include it everywhere + the signature is overwritten by the one in cuhxx because it is included after reduce.cuh